poly-synth/src/poly-synth-demo.js

```diff
@@ -22,163 +22,368 @@ class PolySynthDemo {
-  }
-
-  /**
-   * Initialize GUI components.
    * @param {String} gainADSRId The ID of the container for gain ADSR elements
    *                            (e.g. <div>).
    * @param {String} filterADSRId The ID of the container for filter
    *                              ADSR elements.
+   * @param {String} bitcrusherId The ID of the container for bitcrusher
+   *                              parameter elements.
+   * @param {String} convolverId The ID of the container for convolver
+   *                             elements.
+   * @param {String} convolverSelectorId The ID of the <select> element for
+   *                             convolver impulse response options.
+   * @param {String} feedbackId The ID of the container for feedback delay
+   *                             elements.
    */
-  initializeGUI(gainADSRId, filterADSRId) {
+  constructor(
+      context, gainADSRId, filterADSRId, bitcrusherId, convolverId,
+      convolverSelectorId, feedbackId) {
+    this.context_ = context;
+    this.gainADSRId_ = gainADSRId;
+    this.filterADSRId_ = filterADSRId;
+    this.bitcrusherId_ = bitcrusherId;
+    this.convolverId_ = convolverId;
+    this.convolverSelectorId_ = convolverSelectorId;
+    this.feedbackId_ = feedbackId;
+
+    this.impulseResponseUrls_ = this.getImpulseResponseUrls_();
+
+    this.masterGain_ = new GainNode(this.context_, {gain: 0.5});
+
+    this.loadImpulseResponses(this.impulseResponseUrls_)
+        .then((impulseResponseBuffers) => {
+          this.impulseResponseBuffers_ = impulseResponseBuffers;
+          
+          this.polySynth_ =
+              new PolySynth(this.context_, impulseResponseBuffers[0]);
+          this.polySynth_.output.connect(this.masterGain_)
+              .connect(this.context_.destination);
+
+          // The QWERTY keyboard is defined in qwerty-hancock.min.js.
+          this.keyboard_ = new QwertyHancock(
+              {id: 'keyboard', width: 600, height: 150, octaves: 2});
+          this.keyboard_.keyDown = this.keyDown.bind(this);
+          this.keyboard_.keyUp = this.keyUp.bind(this);
+          
+          this.initializeAmplifierEnvelopeGUI_(this.gainADSRId_);
+          this.initializeFilterEnvelopeGUI_(this.filterADSRId_);
+          this.initializeBitcrusherGUI_(this.bitcrusherId_);
+          this.initializeConvolverGUI_(
+              this.convolverId_, this.convolverSelectorId_);
+          this.initializeFeedbackDelayGUI_(this.feedbackId_);
+        });
+  }
+
+  async loadImpulseResponses(impulseResponses){
+    let responseBuffers = [];
+    for (let index in impulseResponses) {
+      responseBuffers[index] = await this.loadSound(impulseResponses[index]);
+    }
+    return responseBuffers;
+  }
+
+  async loadSound(url) {
+    const response = await fetch(url);
+    const sound = await response.arrayBuffer();
+    return this.context_.decodeAudioData(sound);
+  }
+
+  initializeAmplifierEnvelopeGUI_(containerId){
     let masterGainSlider_ =
```

Comment:
Not sure why it's difficult to move this out of this method, but it is just a nit. This is not a reusable class, so I am fine with it.